common/src/main/java/whocraft/tardis_refined/common/blockentity/device/AstralManipulatorBlockEntity.java

```diff
@@ -239,10 +240,23 @@ private void attemptToBuild(BlockPos pointA, BlockPos pointB) {
                 List<ItemEntity> droppedItems = getLevel().getEntitiesOfClass(ItemEntity.class, new AABB(pointABlockPos, pointBBlockPos).inflate(1));
                 droppedItems.forEach(Entity::discard);
 
-                // TODO: Make this system also accept placing a structure.
-                ItemStack itemStack = new ItemStack(recipe.recipeOutput.asItem());
-                ItemEntity item = new ItemEntity(level, getBlockPos().getX() + 0.5f, getBlockPos().getY() + 1, getBlockPos().getZ() + 0.5f, itemStack);
-                level.addFreshEntity(item);
+                if (recipe.recipeOutputBlock != null) {
```

Comment:
Use a boolean flag in addition to this null check since we are effectively allowing the recipe to have two different output types. The null check is still good to prevent a hard crash which can occur in the future when recipes are data driven, since users may make a mistake and provide an invalid output in the JSON file.